account_partial_reconcile_restrict/models/account_partial_reconcile.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import ValidationError
+
+
+class AccountPartialReconcile(models.Model):
+    _inherit = "account.partial.reconcile"
+
+    def unlink(self):
+        if not self.user_has_groups("account.group_account_manager"):
```

Comment:
`user_has_groups()` is more versatile than `user.has_group()`. The latter should suffice.

```suggestion
        if not self.env.user.has_group("account.group_account_manager"):
```